Resources/config/discovery.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="http_client.client.default"
+             class="Http\Adapter\HttpAdapter">
```

Comment:
when we specify a factory, the class is purely for documentation purposes. IDEs like phpstorm with the symfony plugin will be able to code-complete thanks to this.